src/App.js

```diff
@@ -1,16 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const SENDER_1 = chatMessages[0].sender;
+  const SENDER_2 = chatMessages[1].sender;
+
+  const DEFAULT_COLOR_DATA = {
+    [SENDER_1] : '',
+    [SENDER_2]: ''
+  };
+
+  const [colorData, setColorData] = useState(DEFAULT_COLOR_DATA);
+
+  const updateLiked = (messageId) => {
+    const updatedMessages = messages.map(message => {
+      if (messageId === message.id) {
+        return {
+          ...message,
+          liked: !message.liked
+        }
+      }
+
+      return message;
+    });
+    setMessages(updatedMessages);
+  };
+
+  const likeCount = () => {
+    let likeCount = 0;
+    for (const message of messages) {
+      if (message.liked) {
+        likeCount++;
+      }
+    }
+    return likeCount;
+  };
+
+  // Function to pass to ColorChoice to change colorData
+  const setColorCallback = (sender, color) => {
+    const updatedColorData = {...colorData};
+    updatedColorData[sender] = color;
+    setColorData(updatedColorData);
+  };
+
+  // Functions to get colors for each sender
+  const setColorForSender1Callback = (color) => setColorCallback(SENDER_1, color);
+  const setColorForSender2Callback = (color) => setColorCallback(SENDER_2, color);
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          {'Chat Between '}
+          <span className={colorData[SENDER_1]}> {SENDER_1} </span>
+          {' and '}
+          <span className={colorData[SENDER_2]}> {SENDER_2} </span>
+        </h1>
+        <section> 
+          <span className={colorData[SENDER_1]}> {`${SENDER_1}'s color:`} </span>
+          <ColorChoice setColorCallback={setColorForSender1Callback}></ColorChoice>
+
+          <span id="heartWidget" className="widget"> {`${likeCount()} ❤️s liked`} </span>
```

Comment:
You can also write this just as JSX with an embedded JavaScript expression:
```suggestion
          <span id="heartWidget" className="widget">{likeCount()} ❤️s liked</span>
```